### PR TITLE
Send registration email

### PR DESCRIPTION
**Pull request by speasley**

### CODE REVIEW

**Comment by speasley** [comment]
@mokhan Ello, ello, ello.

File: app/views/user_mailer/registration_email.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<html>
+  <body>
```
Comment (mokhan) [inline app/views/user_mailer/registration_email.html.erb]: I think the <html><body> is added by the app/views/layouts/mailer.html.erb. So we may not need it here.

File: app/views/user_mailer/registration_email.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    <p><%= "Hi, #{@username}." %></p>
+    <%= yield %>
+    <p>The Supply Team</p>
+  </body>
+</html>
```
Comment (mokhan) [inline app/views/user_mailer/registration_email.html.erb]: Ditto. Can we check with an [ActionMailer Preview](http://guides.rubyonrails.org/4_1_release_notes.html#action-mailer-previews)

File: app/views/user_mailer/registration_email.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<html>
+  <body>
+    <p><%= "Hi, #{@username}." %></p>
+    <%= yield %>
```
Comment (mokhan) [inline app/views/user_mailer/registration_email.html.erb]: Do we need this yield?

File: app/views/user_mailer/registration_email.text.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<%= "Hi, #{@username}." %>
+
+<%= yield %>
```
Comment (mokhan) [inline app/views/user_mailer/registration_email.text.erb]: Ditto.

File: app/mailers/application_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "from@example.com"
```
Comment (mokhan) [inline app/mailers/application_mailer.rb]: Can we move this to the default [action mailer configuration](http://guides.rubyonrails.org/action_mailer_basics.html#example-action-mailer-configuration)? We can hardcode in test.rb but should probably load from an environment variable in production.rb.

test.rb
```ruby
config.action_mailer.default_options = {from: 'from@example.com'}
```

production.rb
```ruby
config.action_mailer.default_options = { from: ENV.fetch("DEFAULT_FROM_ADDRESS") }
```

File: spec/mailers/user_mailer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    it "renders the recipient email" do
+      expect(mail.to).to eql([user.email])
+    end
+    it "renders the sender email" do
+      expect(mail.from).to eql(["from@example.com"])
+    end
+    it "assigns a username" do
+      expect(mail.body.encoded).to match(user.username)
+    end
+  end
```
Comment (mokhan) [inline spec/mailers/user_mailer_spec.rb]: Nice. :clap:

File: app/controllers/registrations_controller.rb [hunk 7]
```diff
@@ -7,6 +7,7 @@ def create
     @user = User.new(secure_params)
     if @user.save
       log_in(@user)
+      UserMailer.registration_email(@user).deliver_later
```
Comment (mokhan) [inline app/controllers/registrations_controller.rb]: We're going to have to remember to change the [adapter](http://guides.rubyonrails.org/active_job_basics.html#setting-the-backend). For now i guess the default of inline is probably ok.

**Comment by mokhan** [comment]
:+1:

File: spec/mailers/user_mailer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe UserMailer, :type => :mailer do
```
Comment (mokhan) [inline spec/mailers/user_mailer_spec.rb]:
```ruby
type: :mailer
```

**Pull request merged**